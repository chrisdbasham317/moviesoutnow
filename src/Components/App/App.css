.header {
  display: flex;
  justify-content: space-between;
  padding: 20px;
}

.movie_time_logo {
  height: 150px;
}