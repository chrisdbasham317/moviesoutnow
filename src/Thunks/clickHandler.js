import { setCurrentMovie } from '../Actions';

export const clickHandler = (event, id) => {
  return (dispatch) => {
    event.stopPropagation();
    dispatch(setCurrentMovie(id));
  }
}