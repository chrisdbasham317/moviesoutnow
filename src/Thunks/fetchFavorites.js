import { isLoading, hasErrored, setFavorites } from '../Actions'

export const fetchFavorites = (id) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(`http://localhost:3001/api/v1/users/${id}/moviefavorites`)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      const favoritesResponse = await response.json() 
      dispatch(isLoading(false));
      dispatch(setFavorites(favoritesResponse.favorites));
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}