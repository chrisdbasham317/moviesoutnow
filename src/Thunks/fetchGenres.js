import { isLoading, hasErrored, setGenres } from '../Actions';

export const fetchGenres = () => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch('https://api.themoviedb.org/3/genre/movie/list?api_key=02dd2ef67fc6cb12ff710ae75f51dda5&language=en-US')
      if (!response.ok) {
        throw Error(response.statusText)
      }
      const genres = await response.json();
      dispatch(isLoading(false));
      dispatch(setGenres(genres.genres));
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}