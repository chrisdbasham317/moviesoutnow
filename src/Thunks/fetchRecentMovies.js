import { isLoading, hasErrored, setRecentMovies } from '../Actions';



export const fetchRecentMovies = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      const movies = await response.json();
      dispatch(isLoading(false));
      dispatch(setRecentMovies(movies));
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}