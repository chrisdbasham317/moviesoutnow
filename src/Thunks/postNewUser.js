import { isLoading, hasErrored, currentUser } from '../Actions'

export const postNewUser = (loginCredentials) => {
  return async (dispatch) => {
    const { firstName, lastName, email, password } = loginCredentials
    const formattedUser = {
      name: firstName + " " + lastName,
      email,
      password
    }
    const options = {
      method: 'POST',
      body: JSON.stringify(formattedUser),
      headers: {
        'Content-Type': 'application/json'
      }
    };
    try {
      dispatch(isLoading(true))
      const response = await fetch(`http://localhost:3001/api/v1/users/`, options)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      const newUserResponse = await response.json()
      dispatch(isLoading(false));
      dispatch(currentUser(newUserResponse));
      dispatch(hasErrored('', { type: 'CLEAR_ERROR' }))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}