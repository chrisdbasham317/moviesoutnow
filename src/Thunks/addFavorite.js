import { isLoading, hasErrored, setFavorites } from '../Actions'
import { fetchFavorites } from './fetchFavorites';


export const addFavorite = (event, userId, movieInfo) => {
  event.preventDefault()
  const { id, title, poster_path, release_date, vote_average, overview } = movieInfo;
  const cleanedInfo = {
    movie_id: id,
    title,
    poster_path,
    release_date,
    vote_average,
    overview
  }
  const options = {
    method: 'POST',
    body: JSON.stringify(cleanedInfo),
    headers: {
      'Content-Type': 'application/json'
    }
  };
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(`http://localhost:3001/api/v1/users/${userId}/moviefavorites`, options)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false));
      dispatch(fetchFavorites(userId))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}