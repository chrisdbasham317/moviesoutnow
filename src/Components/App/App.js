import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchRecentMovies } from '../../Thunks/fetchRecentMovies';
import { fetchGenres } from '../../Thunks/fetchGenres';
import { bindActionCreators } from 'redux';
import Movies from '../../Containers/Movies/Movies';
import './App.css';

class App extends Component {

  componentDidMount() {
    this.props.fetchRecentMovies('https://api.themoviedb.org/3/movie/now_playing?api_key=02dd2ef67fc6cb12ff710ae75f51dda5&language=en-US&page=1')
    this.props.fetchGenres()
  }

  render() {
    return (
      <div>
        <h1>Hello!</h1>
        <Movies />
      </div>  
    )
  }
}


  const mapStateToProps = ({ movies, genres, isLoading, error }) => ({
    movies,
    genres,
    isLoading,
    error
  })

const mapDispatchToProps = (dispatch) => (
  bindActionCreators({fetchRecentMovies, fetchGenres}, dispatch)
)




export default connect(mapStateToProps, mapDispatchToProps)(App)