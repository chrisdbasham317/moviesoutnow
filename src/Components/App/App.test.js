import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { fetchGenres } from '../../Thunks/fetchGenres';
import { fetchRecentMovies } from '../../Thunks/fetchRecentMovies';

jest.mock('../../Thunks/fetchGenres');
jest.mock('../../Thunks/fetchRecentMovies');

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <App
        movies={[
          {
            popularity: 458.719,
            vote_count: 4305,
            video: false,
            poster_path: "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
            id: 475557,
            adult: false,
            backdrop_path: "/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg",
            original_language: "en",
            original_title: "Joker",
            genre_ids: [
              80,
              18,
              53
            ],
            title: "Joker",
            vote_average: 8.6,
            overview: "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
            "release_date": "2019-10-04"
          },
          {
            popularity: 231.895,
            vote_count: 472,
            video: false,
            poster_path: "/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg",
            id: 420809,
            adult: false,
            backdrop_path: "/skvI4rYFrKXS73BJxWGH54Omlvv.jpg",
            original_language: "en",
            original_title: "Maleficent: Mistress of Evil",
            genre_ids: [
              12,
              14,
              10751
            ],
            title: "Maleficent: Mistress of Evil",
            vote_average: 7.2,
            overview: "Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.",
            "release_date": "2019-10-18"
          }
        ]}
        genres={[{
          id: 30,
          name: 'Action'
        },
        {
          id: 44,
          name: 'Comedy'
        },
        {
          id: 32,
          name: 'Horror'
        }]}
        isLoading={false}
        error=''
        toggleModal={true}
        currentUser={{}}
        fetchRecentMovies={jest.fn()}
        fetchGenres={jest.fn()}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls dispatch with fetchGenres when componentDidMount is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = fetchGenres();

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchGenres();

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('calls dispatch with fetchRecentMovies when componentDidMount is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = fetchRecentMovies('https://api.themoviedb.org/3/movie/now_playing?api_key=02dd2ef67fc6cb12ff710ae75f51dda5&language=en-US&page=1');

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchRecentMovies('https://api.themoviedb.org/3/movie/now_playing?api_key=02dd2ef67fc6cb12ff710ae75f51dda5&language=en-US&page=1');

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  describe('mapStateToProps', () => {
    it('should return and object with the redux state', () => {
      const mockState = {
        movies: [{ title: 'movie1' }, { title: 'movie2' }],
        currentUser: {},
        error: '',
        genres: [{ name: 'genre1' }],
        isLoading: false,
        toggleModal: true
      };
      const expected = mockState

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});
