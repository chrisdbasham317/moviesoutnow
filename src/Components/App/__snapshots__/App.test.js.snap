// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <div
      className="logo--homebtn__div"
    >
      <Link
        to="/"
      >
        <img
          alt="Logo"
          className="movie_time_logo"
          src="movie_time.png"
        />
      </Link>
      <Link
        to="/"
      >
        <button
          className="go-home__button"
          type="button"
        >
          Home
        </button>
      </Link>
    </div>
    <Connect(LoginForm) />
  </header>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    className="SignupFormModal"
    closeTimeoutMS={0}
    contentLabel="Signup Form"
    isOpen={true}
    overlayClassName="SignupFormOverlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {},
        "overlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.85)",
          "bottom": 0,
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
        },
      }
    }
  >
    <img
      alt="Cinema Night Art Work"
      className="cinema_night_artwork"
      src="cinema-night.png"
    />
    <Connect(SignupForm) />
  </Modal>
  <Route
    exact={true}
    path="/"
  >
    <Connect(Movies) />
  </Route>
  <Route
    path="/movies/:id"
  >
    <Connect(MoviePage) />
  </Route>
  <Route
    exact={true}
    path="/favorites"
  >
    <Connect(Favorites) />
  </Route>
</div>
`;
